lib/tapioca/sorbet_ext/generic_name_patch.rb

```diff
@@ -13,29 +13,35 @@ def [](*types)
 
         return instances[instance_key] if instances.key?(instance_key)
 
-        singleton_class.prepend(Module.new { define_method(:name) { name } })
+        T.unsafe(self).singleton_class.prepend(Module.new { T.unsafe(self).define_method(:name) { name } })
         instances[instance_key] = self
       end
 
       def type_member(variance = :invariant, fixed: nil, lower: T.untyped, upper: BasicObject)
         super.tap do |type_member|
           type_member.singleton_class.instance_exec do
-            define_method(:fixed) { fixed }
-            define_method(:lower) { lower unless lower == T.untyped }
-            define_method(:upper) { upper unless upper == BasicObject }
+            T.unsafe(self).define_method(:fixed) { fixed }
+            T.unsafe(self).define_method(:lower) { lower unless lower == T.untyped }
+            T.unsafe(self).define_method(:upper) { upper unless upper == BasicObject }
```

Comment:
Is `unless upper == BasicObject` to avoid printing the default values for those bounds in the generated RBI?

Reply:
Yes, `BasicObject` is the default value of `upper` and `T.untyped` is the default value of `lower`, so, I am not generating a `lower:`/`upper:` definition if those are the values to keep the noise levels down in the generated RBI file.